wasm/src/lsp/types.rs

```diff
@@ -111,6 +113,14 @@ impl HasFilePath for Url {
     }
 }
 
+pub trait IFileReader {
+    fn read(&self, name: &str) -> Result<Vec<u8>, String>;
+}
+
+pub trait ILogWriter {
+    fn write(&self, text: &str);
```

Comment:
I think `write()` returns `Result` type indicating whether write operation succeeds.
e.g. `write` function in `std::io::Write` trait returns `Result<usize>`. (Returning how many bytes were successfully written)

Reply:
I didn't make log return a result because it's not critical for it to succeed.  This is provided by the javascript cradle (or by a simple implementation in rust using log or eprintln!.

Maybe I should rename it to be "log" or something here...

Reply:
Renamed both and made the result more clear.